app/graphql/foreman_graphql_schema.rb

```diff
@@ -19,10 +19,8 @@ def self.id_from_object(object, type_definition, query_ctx)
   def self.object_from_id(id, query_ctx)
     return unless id.present?
 
-    _, type_name, item_id = Foreman::GlobalId.decode(id)
-    type_class = "::Types::#{type_name}".safe_constantize
```

Comment:
it won't recognize the type from the plugin namespace, eg. `ForemanMyPlugin::Types::MyType`

Reply:
I believe this was added as a security feature. The global id is user input and I believe it's not recommended to call `constantize` on user input. Can we maybe filter this to only allow classes that are Types?

Reply:
good catch, what do you think about `ForemanGraphqlSchema.types.keys.find { |t| t == type_name }&.safe_constantize`?

Reply:
This looks good.